diff --git a/node_modules/react-native-screens/android/CMakeLists.txt b/node_modules/react-native-screens/android/CMakeLists.txt
index 1234567..abcdefg 100644
--- a/node_modules/react-native-screens/android/CMakeLists.txt
+++ b/node_modules/react-native-screens/android/CMakeLists.txt
@@ -41,12 +41,14 @@ find_package(ReactAndroid REQUIRED CONFIG)
 if(${RNS_NEW_ARCH_ENABLED})
     find_package(fbjni REQUIRED CONFIG)
     target_link_libraries(rnscreens
         ReactAndroid::reactnative
         ReactAndroid::jsi
         fbjni::fbjni
         android
+        c++_shared
     )
 else()
     target_link_libraries(rnscreens
         ReactAndroid::jsi
         android
+        c++_shared
     )
 endif()
